docs/concepts/projects.md

```diff
@@ -24,89 +24,225 @@ version = "0.1.0"
 description = "Add your description here"
 ```
 
-It's recommended, but not required, to include a Python version requirement:
+It's recommended, but not required, to include a Python version requirement in the `[project]`
+section:
 
 ```toml title="pyproject.toml"
-[project]
 requires-python = ">=3.12"
 ```
 
-This Python version requirement determines what syntax is valid in the project and affects the
-versions of dependencies which can be used (they must support the same Python range).
+Including a Python version requirement defines what Python syntax is allowed in the project and
+affects selection of dependency versions (they must support the same Python version range).
 
-The `pyproject.toml` also lists dependencies of the project. uv supports modifying the standard
-dependency list from the command line with `uv add` and `uv remove`. uv also supports
-[extended package sources](./dependencies.md) for advanced users.
+The `pyproject.toml` also lists dependencies of the project in the `project.dependencies` and
+`project.optional-dependencies` fields. uv supports modifying the project's dependencies from the
+command line with `uv add` and `uv remove`. uv also supports extending the standard dependency
+definitions with [package sources](./dependencies.md) in `tool.uv.sources`.
 
 !!! tip
 
     See the official [`pyproject.toml` guide](https://packaging.python.org/en/latest/guides/writing-pyproject-toml/) for more details on getting started with a `pyproject.toml`.
 
+## Build systems
+
+Projects _may_ define a `[build-system]` in the `pyproject.toml`. The build system defines how the
+project should be packaged and installed.
+
+uv uses the presence of a build system to determine if a project contains a package that should be
+installed in the project virtual environment. If a build system is not defined, uv will not attempt
+to build or install the project itself, just its dependencies. If a build system is defined, uv will
+build and install the project (as an editable) into the project environment.
```

Comment:
Should we briefly state what the implication of "as an editable" is? That it will update with your project's sources, basically?

Reply:
I'm sort of regretting mentioning it at all. I feel like we need a dedicated section for that. I'll do something brief here though.